File: Examples/CSharp/Tables/CreateTableWithLockedColumns.cs
using System;

namespace Aspose.Note.Examples.CSharp.Tables
{
    public class CreateTableWithLockedColumns
    {
        public static void Run()
        {
            // ExStart:CreateTableWithLockedColumns
            // ExFor:Table
            // ExFor:Table.Columns
            // ExFor:Table.IsBordersVisible
            // ExFor:TableColumn
            // ExFor:TableColumn.Width
            // ExFor:TableColumn.LockedWidth
            // ExFor:TableRow
            // ExFor:TableCell
            // ExSummary:Shows how to create a table with a locked column.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tables();

            // Create an object of the Document class
            Document doc = new Document();

            // Initialize Page class object
            Page page = new Page();

            // Initialize TableRow class object
            TableRow row1 = new TableRow();
            
            // Initialize TableCell class object and set text content
            TableCell cell11 = new TableCell();
            cell11.AppendChildLast(InsertTable.GetOutlineElementWithText("Small text"));
            row1.AppendChildLast(cell11);

            // Initialize TableRow class object
            TableRow row2 = new TableRow();
            
            // Initialize TableCell class object and set text content
            TableCell cell21 = new TableCell();
            cell21.AppendChildLast(InsertTable.GetOutlineElementWithText("Long   text    with    several   words and    spaces."));
            row2.AppendChildLast(cell21);

            // Initialize Table class object
            Table table = new Table()
                          {
                              IsBordersVisible = true,
                              Columns = { new TableColumn { Width = 70, LockedWidth = true } }
                          };
            
            // Add rows
            table.AppendChildLast(row1);
            table.AppendChildLast(row2);

            Outline outline = new Outline();
            OutlineElement outlineElem = new OutlineElement();

            // Add table node
            outlineElem.AppendChildLast(table);
            
            // Add outline element node
            outline.AppendChildLast(outlineElem);
            
            // Add outline node
            page.AppendChildLast(outline);
            
            // Add page node
            doc.AppendChildLast(page);
            dataDir = dataDir + "CreateTableWithLockedColumns_out.one";
            doc.Save(dataDir);
            
            // ExEnd:CreateTableWithLockedColumns

            Console.WriteLine("\nTable with locked columns created successfully.\nFile saved at " + dataDir);
        }
        
    }
}

File: Examples/VisualBasic/Tables/CreateTableWithLockedColumns.vb
Imports System.IO
Imports  Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing

Namespace Tables
    Public Class CreateTableWithLockedColumns
        Public Shared Sub Run()
            ' ExStart:CreateTableWithLockedColumns
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tables()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New Page(doc)

            ' Initialize TableRow class object
            Dim row1 As New TableRow(doc)
            ' Initialize TableCell class object and set text content
            Dim cell11 As New TableCell(doc)
            cell11.AppendChild(InsertTable.GetOutlineElementWithText(doc, "Small text"))
            row1.AppendChild(cell11)

            ' Initialize TableRow class object
            Dim row2 As New TableRow(doc)
            ' Initialize TableCell class object and set text content
            Dim cell21 As New TableCell(doc)
            cell21.AppendChild(InsertTable.GetOutlineElementWithText(doc, "Long   text    with    several   words and    spaces."))
            row2.AppendChild(cell21)

            ' Initialize Table class object
            Dim table As New Table(doc) With {.IsBordersVisible = True}
            ' Add rows
            Table.AppendChild(row1)
            Table.AppendChild(row2)

            Dim outline As New Outline(doc)
            Dim outlineElem As New OutlineElement(doc)
            ' Add table node
            outlineElem.AppendChild(Table)
            ' Add outline element node
            outline.AppendChild(outlineElem)
            ' Add outline node
            page.AppendChild(outline)
            ' Add page node
            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("CreateTableWithLockedColumns_out.one")

            'doc.Save(dataDir)
            ' ExEnd:CreateTableWithLockedColumns

            Console.WriteLine(Convert.ToString(vbLf & "Table with locked columns created successfully." & vbLf & "File saved at ") & dataDir)
        End Sub

    End Class
End Namespace